src/pytorch_lightning/loggers/comet.py

```diff
@@ -141,14 +141,16 @@ def __init__(
         prefix: str = "",
         agg_key_funcs: Optional[Mapping[str, Callable[[Sequence[float]], float]]] = None,
         agg_default_func: Optional[Callable[[Sequence[float]], float]] = None,
-        **kwargs,
+        **kwargs: Any,
```

Comment:
@mauvilsa We are setting the type for the kwargs here. However, then the test in `test_cli.py::test_comet_logger_init_args` fails with
```

E               ValueError: Value "Namespace(class_path='pytorch_lightning.loggers.CometLogger', init_args=Namespace(save_dir='comet', workspace='comet'))" does not validate against any of the types in typing.Union[pytorch_lightning.loggers.logger.Logger, typing.Iterable[pytorch_lightning.loggers.logger.Logger], bool]:
E                 - expected AST, got 'NoneType'
E                 - Expected a List but got "Namespace(class_path='pytorch_lightning.loggers.CometLogger', init_args=Namespace(save_dir='comet', workspace='comet'))"
E                 - Expected a <class 'bool'> but got "Namespace(class_path='pytorch_lightning.loggers.CometLogger', init_args=Namespace(save_dir='comet', workspace='comet'))"
```

https://github.com/Lightning-AI/lightning/runs/7476365010?check_suite_focus=true

Do you know what the correct annotation is here? Perhaps we should drop it and ignore the mypy error?

Reply:
The issue is not the type of `**kwargs` but a bug in jsonargparse and the

Reply:
The issue is not the type of `**kwargs` but a bug in jsonargparse and the `self._save_dir: Optional[str]`, `self.rest_api_key: Optional[str]` lines below. This bug was fixed in https://github.com/omni-us/jsonargparse/pull/148 and has already been released. But with the new version a test will fail. I created a pull request to fix the test https://github.com/Lightning-AI/lightning/pull/13805.

Reply:
Thank you!